### PR TITLE
General changes to surveys

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

[Minor changes to surveys pages](https://github.com/webkom/lego-webapp/commit/1b0b1558dd73e7769068127ff5c5eb6a3d2829c3) 

Add a "cancel" button to the editor and a "back" navigator.
Other than that just styling changes.

---
  
[Rewrite survey editor to a functional component](https://github.com/webkom/lego-webapp/commit/745180a8e33f5c1d41ca4f032a2b95c2b6a1e59c)

---

[Migrate survey editor to react-final-form](https://github.com/webkom/lego-webapp/commit/a9f0f246b0ac11b00bf768dbf8a3ec687c7abd7a)

# Testing

- [x] I have thoroughly tested my changes.

The changes to the `<Field />` component has been checked with fields in `redux-form` forms using the `onChange` prop.

Editing works for the three different options; radio buttons, check boxes and text areas.

---

Related to ABA-54 (`redux-form` is still used on the actual survey form)

### CODE REVIEW

File: app/routes/surveys/components/SurveyEditor/Question.tsx [hunk 1]
```diff
@@ -16,18 +17,15 @@ import {
...
 type Props = {
   deleteQuestion: (arg0: number) => Promise<any>;
   questionData: Record<string, any>;
   question: string;
   relativeIndex: number;
   numberOfQuestions: number;
-  updateRelativeIndexes: (arg0: number, arg1: number, arg2: Fields) => void;
-  fields: Fields;
+  updateRelativeIndexes: (arg0: number, arg1: number, arg2: any) => void;
+  fields: any;
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/Question.tsx:28]: :cry:

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx [hunk 2]
```diff
@@ -1,47 +1,44 @@
...
-type Props = FormProps & {
+type Props = {
   survey: SurveyEntity;
   autoFocus: Record<string, any>;
   surveyData: Array<Record<string, any>>;
-  submitFunction: (
-    arg0: SurveyEntity,
-    arg1: number | null | undefined
-  ) => Promise<any>;
+  submitFunction: (arg0: any) => Promise<any>;
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:34]: `void`?

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx [hunk 3]
```diff
@@ -1,47 +1,44 @@
...
   template?: Record<string, any>;
   selectedTemplateType?: EventType;
   destroy: () => void;
   initialize: () => void;
   activeFrom: string;
-};
-type State = {
-  templatePickerOpen: boolean;
-  templateTypeSelected: string;
+  initialValues: Record<string, any>;
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:41]: :cry:

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx [hunk 4]
```diff
@@ -61,7 +58,7 @@ function TemplateTypeDropdownItems({
 
   return (
     <Dropdown.List>
-      {Object.keys(EVENT_CONSTANTS).map((eventType) => (
+      {Object.keys(EVENT_CONSTANTS).map((eventType: EventType) => (
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:61]: wanna make this an enum? :grimacing:

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx [hunk 5]
```diff
@@ -287,6 +107,46 @@ export const hasOptions = (data: Record<string, any>) => {
...
+const renderQuestions = ({
+  fields,
+  meta: { touched, error },
+}: any): ReactNode => {
+  return (
+    <>
+      <ul className={styles.questions} key="questions">
+        {fields.map((question, i) => (
+          <Question
+            key={i}
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:123]: :thinking:

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx [hunk 6]
```diff
@@ -295,41 +155,193 @@ const validate = createValidator(
...
+      return question;
+    });
+    return submitFunction({
+      ...formContent,
+      surveyId: survey?.id,
+      title: formContent.title,
+      event: formContent.event && Number(formContent.event.value),
+      questions: cleanQuestions,
+    }).then((result) => {
+      const id = survey && survey.id ? survey.id : result.payload.result;
```
Comment (LudvigHz) [inline app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:198]: `survey?.id`

**Review by LudvigHz — APPROVED** [review]
LGTM
Types could be better, otherwise just small comments

File: app/routes/surveys/components/SurveyEditor/Question.tsx (hunk 1 above)
Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/Question.tsx:28]: These are so difficult to type 😩

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx (hunk 2 above)
Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:34]: Probably

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx (hunk 3 above)
Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:41]: Do you know how big that object is? I'm not typing that

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx (hunk 6 above)
Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:198]: Sure, but again, not me

File: app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx (hunk 4 above)
Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:61]: Kind of out of scope though, `EVENT_CONSTANTS` is used a lot of places.

Comment (ivarnakken) [reply app/routes/surveys/components/SurveyEditor/SurveyEditor.tsx:61]: Sure

**Pull request merged**